--- location_views_handler_field_latitude.inc.orig	Sun Sep 25 21:42:30 2011
+++ location_views_handler_field_latitude.inc	Sun Sep 25 21:44:48 2011
@@ -19,7 +19,10 @@
     $form['style'] = array(
       '#title' => t('Display style'),
       '#type' => 'select',
-      '#options' => array('dd' => t('Decimal degrees'), 'dms' => t('Degrees, minutes, seconds')),
+      //begin patch by greenmap opengreenmap openflows mark libkuman
+      //'#options' => array('dd' => t('Decimal degrees'), 'dms' => t('Degrees, minutes, seconds')),
+      '#options' => array('dd' => t('Decimal degrees'), 'dms' => t('Degrees, minutes, seconds'), 'kml' => t('Decimal degrees reverse order for kml')),
+      //end patch by greenmap opengreenmap openflows mark libkuman
       '#default_value' => $this->options['style'],
     );
   }
