--- location_views_handler_field_coordinates.inc.orig	Sun Sep 25 21:42:13 2011
+++ location_views_handler_field_coordinates.inc	Sun Sep 25 21:39:54 2011
@@ -17,6 +17,11 @@
     if ($this->options['style'] == 'dms') {
       return theme('location_latitude_dms', $values->{$this->field_alias}) .', '. theme('location_longitude_dms', $values->{$this->aliases['longitude']});
     }
+    //begin patch by greenmap opengreenmap openflows mark libkuman
+    elseif ($this->options['style'] == 'kml') {
+      return check_plain($values->{$this->aliases['longitude']}) .', '. check_plain($values->{$this->field_alias});
+    }
+    //end patch by greenmap opengreenmap openflows mark libkuman
     else {
       return check_plain($values->{$this->field_alias}) .', '. check_plain($values->{$this->aliases['longitude']});
     }
