--- C:\DOCUME~1\THOMAS~1\LOCALS~1\Temp\photobucket.inc-rev88.svn000.tmp.inc	Wed Jan 14 10:24:45 2009
+++ C:\xampp\htdocs\opengreenmap\sites\all\modules\drupal-contrib\emfield\contrib\image_ncck\providers\photobucket.inc	Wed Jan 14 10:15:50 2009
@@ -115,6 +115,8 @@
     return $title[$url];
   }
 
-  $html = file_get_contents($url);
+  // $html = file_get_contents($url);
+  $rs = drupal_http_request($url);
+  $html = $rs->data;
   return $title[$url] = preg_match('@<span id="photoTitle">(.+?)</span>@is', $html, $matches)? $matches[1] : '';
 }
