--- location.module.orig	Mon Sep 26 16:04:45 2011
+++ location.module	Mon Sep 26 16:04:14 2011
@@ -1263,7 +1263,12 @@
   }
   foreach ($location as $k => $v) {
     if (!isset($tmp[$k])) {
-      unset($location[$k]);
+      //begin patch by opengreenmaps greenmap openflows mark libkuman
+      //unset($location[$k]);
+      if(!($location[$k] == "locpick" || $location[$k] == "user_latitude" || $location[$k] == "user_longitude")) {
+        unset($location[$k]);
+      }
+      //end patch by opengreenmaps greenmap openflows mark libkuman
     }
   }
 }
@@ -1313,7 +1318,12 @@
   if (!$inhibit_geocode) {
     // Have any of the fields possibly affecting geocoding changed?
     // Or, was the location previously user submitted but is no longer?
-    if ($changed['street'] || $changed['additional'] || $changed['city'] || $changed['province'] || $changed['country'] || $changed['postal_code'] || $location['source'] == LOCATION_LATLON_USER_SUBMITTED) {
+    
+    //begin patch by opengreenmaps greenmap openflows mark libkuman
+    //if ($changed['street'] || $changed['additional'] || $changed['city'] || $changed['province'] || $changed['country'] || $changed['postal_code'] || $location['source'] == LOCATION_LATLON_USER_SUBMITTED) {
+     if ($changed['street'] || $changed['additional'] || $changed['city'] || $changed['province'] || $changed['country'] || $changed['postal_code'] || $location['source'] == LOCATION_LATLON_USER_SUBMITT
+ED || $location['source'] == LOCATION_LATLON_UNDEFINED) {
+     //end patch by opengreenmaps greenmap openflows mark libkuman
 
       // Attempt exact geocoding.
       if ($data = location_latlon_exact($location)) {
