--- C:\DOCUME~1\THOMAS~1\LOCALS~1\Temp\taxonomy_image.module-rev98.svn000.tmp.module	Tue Oct 28 11:35:06 2008
+++ C:\xampp\htdocs\opengreenmap\sites\all\modules\drupal-contrib\taxonomy_image\taxonomy_image.module	Mon Oct 27 17:56:27 2008
@@ -89,12 +89,12 @@
 
         default:
           if ($preset == 'ORIGINAL') {
-            $return_url = theme('image', $current->url, $current->name, $current->title, $attributes, FALSE);
+            $return_url = theme('image', $current->url, t($current->name), t($current->title), $attributes, FALSE);
           }
           else {
             // Make sure the attributes don't try to override the preset.
             unset($attributes['width'], $attributes['height']);
-            $return_url = theme('imagecache', $preset, $current->path, $current->name, $current->title, $attributes);
+            $return_url = theme('imagecache', $preset, $current->path, t($current->name), t($current->title), $attributes);
           }
       }
       if ($wrapper) {
