--- i18nviews.views.inc.orig	Sun Sep 25 20:42:44 2011
+++ i18nviews.views.inc	Sun Sep 25 20:44:22 2011
@@ -24,8 +24,39 @@
       'content_negotiation_filter_handler' => array(
         'parent' => 'views_handler_filter',
       ),
+      //begin patch by opengreemaps greenmaps openflows mark libkuman
+      //this is for the i18ntaxonomyviews module
+      'i18nviews_handler_field_taxonomy' => array(
+        'parent' => 'views_handler_field_taxonomy',
+      ),
+      'i18nviews_handler_field_term_description' => array(
+        'parent' => 'views_handler_field_markup',
+      ),
+      'i18nviews_handler_field_term_node_tid' => array(
+        'parent' => 'views_handler_field_term_node_tid',
+      ),
+      'i18nviews_handler_argument_term_node_tid' => array(
+        'parent' => 'views_handler_argument_term_node_tid',
+      ),
+      'i18nviews_handler_filter_term_node_tid' => array(
+        'parent' => 'views_handler_filter_term_node_tid',
+      ),
+      //end patch by opengreenmaps greenmaps openflows mark libkuman
     ),
   );
+}
+
+/**
+ * this function patched in for support of the i18ntaxonomyviews module
+ * opengreenmaps greenmap openflows mark libkuman
+ * Implemenation of hook_views_data_alter().
+ */
+function i18nviews_views_data_alter(&$data) {
+  $data['term_data']['name']['field']['handler'] = 'i18nviews_handler_field_taxonomy';
+  $data['term_data']['description']['field']['handler'] = 'i18nviews_handler_field_term_description';
+  $data['term_node']['tid']['field']['handler'] = 'i18nviews_handler_field_term_node_tid';
+  $data['term_node']['tid']['argument']['handler'] = 'i18nviews_handler_argument_term_node_tid';
+  $data['term_node']['tid']['filter']['handler'] = 'i18nviews_handler_filter_term_node_tid';
 }
 
 /**
