--- admin_message.module	Wed Oct 22 13:21:35 2008
+++ admin_message.module	Wed Oct 22 13:12:52 2008
@@ -119,6 +119,23 @@
       // Insert JavaScript and CSS if messages are displayed.
       drupal_add_js(drupal_get_path('module', 'admin_message') .'/admin_message.js', 'module');
       drupal_add_css(drupal_get_path('module', 'admin_message') .'/admin_message.css', 'module');
+	  if(module_exists('messagefx')){
+		    switch (variable_get('status_effect', 'Pulsate')) {
+		      case 'Pulsate' :
+		        $js = "\$('.admin-message').Pulsate(". variable_get('status_length', 200) .", ". variable_get('status_times', 6) .");";
+		        break;
+		      case 'Shake' :
+		        $js = "\$('.admin-message').Shake(". variable_get('status_times', 5) .");";
+		        break;
+		      case 'Highlight' :
+		        $js = "\$('.admin-message').Highlight(". variable_get('status_length', 500) .", '". variable_get('status_highlight_color', '#FF3') ."');";
+		        break;
+		    }
+			if ($js) {
+		      $js = '$(window).load(function(){'. $js .'});';
+		      drupal_add_js($js, 'inline', 'header');
+		    }
+	  }
     }
   }
   return $output;